src/csync/csync_update.cpp

```diff
@@ -762,45 +751,44 @@ int csync_ftw(CSYNC *ctx, const char *uri, unsigned int depth) {
     }
 
     bool recurse = dirent->type == ItemTypeDirectory;
-    csync_file_stat_t *previous_fs = ctx->current_fs;
-    QScopedValueRollback<csync_file_stat_t *> rollback(ctx->current_fs);
 
     /* Call walker function for each file */
+    auto fs = dirent.get(); // Will only stay valid if _csync_detect_update returns 0.
     int rc = _csync_detect_update(ctx, std::move(dirent));
-    /* this function may update ctx->current and ctx->read_from_db */
 
     if (rc < 0) {
       if (CSYNC_STATUS_IS_OK(ctx->status_code)) {
           ctx->status_code = CSYNC_STATUS_UPDATE_ERROR;
       }
       goto error;
     }
+    if (rc != 0) {
+        ctx->remote.read_from_db = read_from_db;
+        continue;
+    }
```

Comment:
Why is that new here?

Reply:
If rc is > 0 , the 'fs' pointer will be dangling, so we need to continue and avoid the next operation that touches 'fs'.

The read_from_db thingy is there to keep the invariant that was there before. This was like that because in the past, it was the vio code that did the read from the db.  
Anyway, this goes away in the next commit.